### ISSUE TITLE
Intermittent socket timeouts

### ISSUE THREAD

**Issue by vanpelt** [wandb/client#660]
This happens rarely but we should likely catch the timeout.

```
"/Midgard/home/mrabadan/anaconda3/envs/pytorch/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
self.run()
File "/Midgard/home/mrabadan/anaconda3/envs/pytorch/lib/python3.7/multiprocessing/process.py", line 99, in run
self._target(*self._args, **self._kwargs)
File "scripts/train_mmnist.py", line 32, in train_run_worker
dir=run_dir)
File "/Midgard/home/mrabadan/anaconda3/envs/pytorch/lib/python3.7/site-packages/wandb/__init__.py", line 983, in init
_init_headless(run)
File "/Midgard/home/mrabadan/anaconda3/envs/pytorch/lib/python3.7/site-packages/wandb/__init__.py", line 239, in _init_headless
success, message = server.listen(30)
File "/Midgard/home/mrabadan/anaconda3/envs/pytorch/lib/python3.7/site-packages/wandb/wandb_socket.py", line 46, in listen
self.connect()
File "/Midgard/home/mrabadan/anaconda3/envs/pytorch/lib/python3.7/site-packages/wandb/wandb_socket.py", line 40, in connect
self.connection, addr = self.socket.accept()
File "/Midgard/home/mrabadan/anaconda3/envs/pytorch/lib/python3.7/socket.py", line 212, in accept
fd, addr = self._accept()
socket.timeout: timed out
```

**Comment by stathius** [comment]
Hi, just started using wandb and I am also getting this problem when using remote nodes and it is not rare at all. Around 1/3 of my jobs end like that. Not sure what is happening but this is leads me to drop wandb for now. Is there a workaround for this?

**Comment by wsjeon** [comment]
I got the exact same issue.

**Comment by vanpelt** [comment]
We're looking into this now.  One thing to try is to set WANDB_DISABLE_CODE=true to see if the timeout still occurs.  @wsjeon if you can provide any more details about your environment that would help us debug.

**Comment by wsjeon** [comment]
@vanpelt Hi, I think what my problem is. In my compute nodes, the internet connection is blocked, and my code includes the `wandb.restore` from the wandb server. I think this is my fault. Thanks for your concern.

**Comment by raubitsj** [comment]
Looks like there is a condition where if we take too long (>30 seconds) to parse your git repo we could hit this condition.  I am going to try and improve this codepath and improve the logging so we can get more information if there is something else hiding.

**Comment by timothybrooks** [comment]
I find `socket.timeout: timed out` in ~half of my runs, even when I set `WANDB_DISABLE_CODE=true`.

```
File "/home/timbrooks/code/adv_rec/utils/logs/logger.py", line 24, in init
    wandb.init(name=name, project=project, dir=log_dir, group=group)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/__init__.py", line 1078, in init
    _init_headless(run)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/__init__.py", line 280, in _init_headless
    success, _ = server.listen(30)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/wandb_socket.py", line 46, in listen
    self.connect()
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/wandb_socket.py", line 40, in connect
    self.connection, addr = self.socket.accept()
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/socket.py", line 212, in accept
    fd, addr = self._accept()
socket.timeout: timed out
(adv_rec) timbrooks@em1:~/code/adv_rec$ Traceback (most recent call last):
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/internal_cli.py", line 106, in <module>
    main()
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/internal_cli.py", line 98, in main
    headless(args)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/internal_cli.py", line 54, in headless
    util.sentry_reraise(e)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/util.py", line 92, in sentry_reraise
    six.reraise(type(exc), exc, sys.exc_info()[2])
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/six.py", line 703, in reraise
    raise value
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/internal_cli.py", line 47, in headless
    port=args['port'])
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/run_manager.py", line 503, in __init__
    self._socket = wandb_socket.Client(self._port)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/wandb_socket.py", line 97, in __init__
    self.socket.connect(('localhost', self.port))
ConnectionRefusedError: [Errno 111] Connection refused
```

**Comment by raubitsj** [comment]
@timothybrooks , what version of wandb were you using. We fixed a race in 0.8.23.

**Comment by timothybrooks** [comment]
0.8.24

```
wandb: Tracking run with wandb version 0.8.24
...
  File "/home/timbrooks/code/adv_rec/utils/logs/logger.py", line 24, in init
    wandb.init(name=name, project=project, dir=log_dir, group=group)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/__init__.py", line 1082, in init
    _init_headless(run)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/__init__.py", line 281, in _init_headless
    success, _ = server.listen(30)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/wandb_socket.py", line 46, in listen
    self.connect()
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/wandb_socket.py", line 40, in connect
    self.connection, addr = self.socket.accept()
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/socket.py", line 212, in accept
    fd, addr = self._accept()
socket.timeout: timed out
(adv_rec) timbrooks@em1:~/code/adv_rec$ Traceback (most recent call last):
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/internal_cli.py", line 106, in <module>
    main()
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/internal_cli.py", line 98, in main
    headless(args)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/internal_cli.py", line 54, in headless
    util.sentry_reraise(e)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/util.py", line 92, in sentry_reraise
    six.reraise(type(exc), exc, sys.exc_info()[2])
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/six.py", line 703, in reraise
    raise value
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/internal_cli.py", line 47, in headless
    port=args['port'])
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/run_manager.py", line 496, in __init__
    self._socket = wandb_socket.Client(self._port)
  File "/home/timbrooks/anaconda3/envs/adv_rec/lib/python3.7/site-packages/wandb/wandb_socket.py", line 97, in __init__
    self.socket.connect(('localhost', self.port))
ConnectionRefusedError: [Errno 111] Connection refused
```

**Comment by raubitsj** [comment]
Can you post the corresponding wandb/debug.log
We added some more debugging in that version which might help us track this down.

**Comment by timothybrooks** [comment]
Sure. Thanks for the help!
```
$ cat debug.log
2020-02-04 16:42:22,944 DEBUG   MainThread:24110 [wandb_config.py:_load_defaults():111] no defaults not found in config-defaults.yaml
2020-02-04 16:42:23,132 DEBUG   MainThread:24110 [connectionpool.py:_new_conn():959] Starting new HTTPS connection (1): api.wandb.ai:443
2020-02-04 16:42:23,285 DEBUG   MainThread:24110 [connectionpool.py:_make_request():437] https://api.wandb.ai:443 "POST /graphql HTTP/1.1" 200 192
2020-02-04 16:42:23,533 DEBUG   raven-sentry.BackgroundWorker:24110 [connectionpool.py:_new_conn():959] Starting new HTTPS connection (1): sentry.io:443
2020-02-04 16:42:23,716 DEBUG   raven-sentry.BackgroundWorker:24110 [connectionpool.py:_make_request():437] https://sentry.io:443 "POST /api/1299483/store/ HTTP/1.1" 200 41
```

**Comment by Yumin-Sun-00** [comment]
I got the same error today. This is already 2nd project I run with wandb. 

I run the project both locally and on a remote server. Locally it is fine, remotely it has time out issue.

I think the problem could because the remote computer spends time assigning tasks, that nearly require 2 minutes or even longer. And maybe this is the reason why wandb return time out error. Could you help to solve this?

**Comment by ZiyunZhao** [comment]
I also got the same error. It works perfectly fine on local but breaks every time when I run on a remote server. Can anyone help please? Really appreciate it!!
<img width="552" alt="Screen Shot 2020-03-06 at 5 18 25 PM" src="https://user-images.githubusercontent.com/42633187/76133830-054ca700-5fcf-11ea-9602-dd168004c537.png">

**Comment by vanpelt** [comment]
@ZiyunZhao @Yumin-Sun-00 My colleague @raubitsj is working on a rewrite of our python client that will address this issue for good.  If you're open to trying it, we should have a beta version next week.  

I'm also happy to do a quick screen share and see if we can live debug the remote environments to get to the root cause.  We've had trouble reproducing this.

Have either of you tried setting WANDB_DISABLE_CODE=true?

**Comment by ZiyunZhao** [comment]
@vanpelt Thank you for the response! I have tried to set WANDB_DISABLE_CODE=true but still not working. I can definitely try the beta version next week and also the live debug.

**Comment by Yumin-Sun-00** [comment]
> @ZiyunZhao @Yumin-Sun-00 My colleague @raubitsj is working on a rewrite of our python client that will address this issue for good. If you're open to trying it, we should have a beta version next week.
> 
> I'm also happy to do a quick screen share and see if we can live debug the remote environments to get to the root cause. We've had trouble reproducing this.
> 
> Have either of you tried setting WANDB_DISABLE_CODE=true?

I can also help for the live debug.

**Comment by djstrong** [comment]
0.8.30 worked for 100 experiments, but now it aborts runs on `wandb.init` with `timed out`. I guess it is Internet connection problem. Should I catch this exception and continue an experiment without wandb?

**Comment by xiaozhoujian** [comment]
I got the exact same issue. I guess the issue is caused by the remote server spends time assigning tasks, in my case, that nearly require 5 minutes or even longer. And maybe this is the reason why wandb return time out error. Since I used the dry run mode of wandb, I just modify the 239 line of __init__.py under the site-packages/wandb as follows:
`# success, _ = server.listen(30)
  success = True`
Then, everything works well. 
It's a temporary solution for those who want to use it now. @Yumin-Sun-00 
Hoping the issue can be solved as soon as  possible.

**Comment by vanpelt** [comment]
This issues should be fixed in our next generation cli.  You can try it by running `pip install wandb-ng`, more information here: https://github.com/wandb/client-ng

**Comment by Xemnas0** [comment]
> This issues should be fixed in our next generation cli. You can try it by running `pip install wandb-ng`, more information here: https://github.com/wandb/client-ng

I tried this and now get the following error

```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/wandb/sdk/wandb_init.py", line 534, in init
    run = wi.init()
  File "/opt/conda/lib/python3.7/site-packages/wandb/sdk/wandb_init.py", line 450, in init
    run._set_run_obj(ret.run)
AttributeError: 'NoneType' object has no attribute 'run'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/wandb/sdk/wandb_init.py", line 539, in init
    wi._atexit_cleanup()
  File "/opt/conda/lib/python3.7/site-packages/wandb/sdk/wandb_init.py", line 317, in _atexit_cleanup
    ret = self.backend.interface.send_exit_sync(exit_code, timeout=60)
AttributeError: 'NoneType' object has no attribute 'interface'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "scripts/run_contrastive.py", line 72, in <module>
    wandb.init(project=args.wandb_proj, tags=tags)
  File "/opt/conda/lib/python3.7/site-packages/wandb/sdk/wandb_init.py", line 552, in init
    raise_from(Exception("problem"), e)
  File "<string>", line 3, in raise_from
Exception: problem
```

**Comment by pyyush** [comment]
Same issue @vanpelt 
`wandb: Tracking run with wandb version 0.9.1
Traceback (most recent call last):
  File "main.py", line 52, in <module>
    wandb.init(project="TransformerTransducer", config=hyperparams) # adds all of the arguments as config variables
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/__init__.py", line 1125, in init
    _init_headless(run)
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/__init__.py", line 293, in _init_headless
    success, _ = server.listen(30)
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/wandb_socket.py", line 48, in listen
    self.connect()
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/wandb_socket.py", line 40, in connect
    self.connection, addr = self.socket.accept()
  File "/N/soft/rhel7/deeplearning/Python-3.7.6/lib/python3.7/socket.py", line 212, in accept
    fd, addr = self._accept()
socket.timeout: timed out
Traceback (most recent call last):
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/internal_cli.py", line 106, in <module>
    main()
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/internal_cli.py", line 98, in main
    headless(args)
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/internal_cli.py", line 54, in headless
    util.sentry_reraise(e)
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/util.py", line 94, in sentry_reraise
    six.reraise(type(exc), exc, sys.exc_info()[2])
  File "/N/soft/rhel7/deeplearning/Python-3.7.6/lib/python3.7/site-packages/six.py", line 703, in reraise
    raise value
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/internal_cli.py", line 47, in headless
    port=args['port'])
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/run_manager.py", line 501, in __init__
    self._socket = wandb_socket.Client(self._port)
  File "/N/u/piyush/Carbonate/.local/lib/python3.7/site-packages/wandb/wandb_socket.py", line 99, in __init__
    self.socket.connect(('localhost', self.port))
ConnectionRefusedError: [Errno 111] Connection refused`

**Comment by pauli31** [comment]
Any solution?

**Comment by vanpelt** [comment]
Hey @pauli31 this should be fixed in the 0.10.x series of the library.  What version of the library are you using and what specific error are you seeing?

**Comment by pauli31** [comment]
> Hey @pauli31 this should be fixed in the 0.10.x series of the library. What version of the library are you using and what specific error are you seeing?

I'm using 0.9.6 
this is the error I'm getting : socket.timeout: timed out


Traceback (most recent call last):
    .............
    wandb.init(project="XXXX",name=args.config_name, config=vars(args), reinit=True)
  File "XXXXX/lib/python3.7/site-packages/wandb/__init__.py", line 1280, in init
    _init_headless(run)
  File "XXXXX/lib/python3.7/site-packages/wandb/__init__.py", line 313, in _init_headless
    success, _ = server.listen(30)
  File "XXXXX/lib/python3.7/site-packages/wandb/wandb_socket.py", line 48, in listen
    self.connect()
  File "XXXXX/lib/python3.7/site-packages/wandb/wandb_socket.py", line 40, in connect
    self.connection, addr = self.socket.accept()
  File "XXXXX/lib/python3.7/socket.py", line 212, in accept
    fd, addr = self._accept()
socket.timeout: timed out

**Comment by pauli31** [comment]
> > Hey @pauli31 this should be fixed in the 0.10.x series of the library. What version of the library are you using and what specific error are you seeing?
> 
> I'm using 0.9.6
> this is the error I'm getting : socket.timeout: timed out
> 
> Traceback (most recent call last):
> .............
> wandb.init(project="XXXX",name=args.config_name, config=vars(args), reinit=True)
> File "XXXXX/lib/python3.7/site-packages/wandb/**init**.py", line 1280, in init
> _init_headless(run)
> File "XXXXX/lib/python3.7/site-packages/wandb/**init**.py", line 313, in _init_headless
> success, _ = server.listen(30)
> File "XXXXX/lib/python3.7/site-packages/wandb/wandb_socket.py", line 48, in listen
> self.connect()
> File "XXXXX/lib/python3.7/site-packages/wandb/wandb_socket.py", line 40, in connect
> self.connection, addr = self.socket.accept()
> File "XXXXX/lib/python3.7/socket.py", line 212, in accept
> fd, addr = self._accept()
> socket.timeout: timed out

I'll try to update the version

**Comment by cvphelps** [comment]
Hi @pauli31 closing for now. Could you ping @ariG23498 if this is still happening and he'll reopen this ticket :)
